### PR TITLE
Add numberOf fields in Imaging Study Resource from dicom-cast

### PR DESCRIPTION
**Pull request by abdullah248**
## Description
Add the numberOfSeries, numberOfInstances, and series.numberOfInstances fields to imaging study resource created by dicom-cast. Based mapping off of DICOM Tag Mapping in the [fhir standards](http://hl7.org/implement/standards/fhir/imagingstudy-mappings.html).

## Related issues
[AB#76993](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/76993)

## Testing
Test cases were added, also validated manually by updating test files to dicom-server and querying imaging study resources from fhir.

### CODE REVIEW

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs [hunk 1]
```diff
@@ -100,5 +102,27 @@ private void AddAccessionNumber(ImagingStudy imagingStudy, DicomDataset dataset)
                 }
             }
         }
+
+        private void AddNumberOfSeries(ImagingStudy imagingStudy, DicomDataset dataset)
+        {
+            if (dataset.TryGetSingleValue(DicomTag.NumberOfStudyRelatedSeries, out int numberOfSeries))
+            {
+                if (imagingStudy.NumberOfSeries == null)
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs:110]: so we would not update it? why? suppose we added more instances and we got a new instance with updated series or instance why would we not updaye it?

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs [hunk 2]
```diff
@@ -100,5 +102,27 @@ private void AddAccessionNumber(ImagingStudy imagingStudy, DicomDataset dataset)
                 }
             }
         }
+
+        private void AddNumberOfSeries(ImagingStudy imagingStudy, DicomDataset dataset)
+        {
+            if (dataset.TryGetSingleValue(DicomTag.NumberOfStudyRelatedSeries, out int numberOfSeries))
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs:108]: for now the worker is running single threaded, how do we handle out of order instance update?

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs (hunk 1 above)
Comment (abdullah248) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs]: Changed to always update so we assume latest uploaded is source of truth.

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudySeriesPropertySynchronizer.cs [hunk 3]
```diff
@@ -38,6 +38,9 @@ public void Synchronize(FhirTransactionContext context, ImagingStudy.SeriesCompo
 
             // Add startedElement to series
             AddStartedElement(series, dataset, context.UtcDateTimeOffset);
+
+            // Add numberOfInstances to series
+            AddNumberOfInstances(series, dataset);
```
Comment (smithago) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudySeriesPropertySynchronizer.cs:43]: I have approved the pr, can you make sure you update the docs to include the additional fields that are synced.

**Review by smithago — APPROVED** [review]

**Comment by abdullah248** [comment]
Closing this pr. Based on feedback we may need to add logic to count instead of a direct mapping.

**Pull request closed without merging**